JASP-Desktop/data/datasetpackage.cpp

```diff
@@ -714,7 +714,9 @@ bool DataSetPackage::setColumnType(int columnIndex, columnType newColumnType)
 	if (_dataSet == nullptr)
 		return true;
 
-	bool changed = _dataSet->column(columnIndex).changeColumnType(newColumnType);
+	bool changed;
+
+	enlargeDataSetIfNecessary([&](){ changed = _dataSet->column(columnIndex).changeColumnType(newColumnType); }, "setColumnType");
```

Comment:
I'm afraid that this enlargeDataSetIfNecessary should be called from other places (Column::setName, Columns:setColumnAsScale...).
Also this enlargeDataSetIfNecessary should throw forward exceptions that are not of type bad_alloc.
But maybe you wanted to get rid of this shared memory, and this fix is temporary?

Reply:
I think if anything breaks inside `enlargeDataSetIfNecessary` we are fucked anyway I think.

Well, this fix is necessary to have JASP not crash in this particular way. I'm sure there are more ways we could make it crash... If someone changes there columnname to something large enough to be too big for the dataset... Or who knows?
But the problem is that I can't really use it inside `Column` or `Label` because those are part of JASP-Common.
And `enlargeDataSetIfNecessary` needs to (possibly) update the pointer to the `DataSet` in `DataSetPackage` which isn't available in JASP-Common.

Definitely part of the reason why I want to get rid of this sharedmemory thing, it is so very fragile.